woocommerce/sensei-certificates
# Fix issues reported by PHPCS

### Changes proposed in this Pull Request

* Fix the main issues reported by PHPCS with the goal to get the WordPress.org plugin approval.

### Testing instructions

* Update the `phpcs.xml.dist` to validate the security questions (it gets the examples that were highlighted by plugins@wordpress.org). Kudos @jom.

```
<?xml version="1.0"?>
<ruleset name="Sensei LMS Certificates">
	<description>A custom set of code standard rules to check for WordPress themes and plugins.</description>
	<arg value="s"/>
	<arg value="p"/>
	<arg name="colors"/>
	<arg name="extensions" value="php"/>
	<file>.</file>
	<exclude-pattern type="relative">^node_modules/*</exclude-pattern>
	<exclude-pattern type="relative">^vendor/*</exclude-pattern>
	<exclude-pattern type="relative">^build/*</exclude-pattern>
	<exclude-pattern type="relative">^lib/*</exclude-pattern>
	<exclude-pattern>tests/</exclude-pattern>
	
	<config name="minimum_supported_wp_version" value="4.9" />
	<config name="testVersion" value="5.6-"/>
	
	<rule ref="WordPress.Security.NonceVerification"/>
	<rule ref="WordPress.Security.ValidatedSanitizedInput" />
	<rule ref="WordPress.Security.EscapeOutput" />
</ruleset>
```
* Run PHPCS and should have no issues.

ba9d1ee813ba3235058c68ace4bb9fbfec22b8b4
- Go to Certificates > Certificate Templates.
- Add a new certificate template.
- Edit some fields in the Certificate Data.
- Save and make sure it saved properly.

bced56bc1deadcd9e4234ee4d2b6eb7fc2ad1a8b
869bf2d8419cb987a6bb95bf9fceaf82249804e3
- Add a new certificate template (or use the same already created).
- Edit the default font fields and default font color field (using the colorpicker).
- Save and make sure it saved properly.

00a4019b2380cd1e7a2e90a9fab90a560f29cae0
1a21615f252f1d33cbff3d3aafdc176b0e06efdb
- Add a new certificate template (or use the same already created).
- Add an image, save, and make sure it saved properly.
- Remove the image, save, and make sure it saved properly.

1a383b111c0d8eb423d66619bca8057b4a7d36f1
- Add a new certificate template (or use the same already created).
- Add a certificate background image.
- Change the positions of the areas on the image.
- Save and make sure it saved properly.
- Remove the positions clicking on the “Remove Position” buttons.
- Save and make sure it saved properly.

ed4ae9b315449c8d29a08953cb1bce7c6b76f7be
- Add a new certificate template (or use the same already created).
- Change all the “font” and “font color” fields for all areas.
- Remove the image, save, and make sure it saved properly.
- Save and make sure it saved properly.

2b31327810845b41f893d12cfc7eb98f42272e2e
- Go to Certificates > Certificate Templates.
- Add a new certificate template.
- Click in preview while it has no image.
- Make sure that the wp_die is fired with the correct error message.

dba5443ba35fafe1a4423f70e9188f30236f52a4
1797080c979a8bb78c1df3c0cab374e7f79892e9
b02a6a8f043119b5fe594e61106ebc4a6fce7e61
a87020784edcea04dc60ea91ce97cbdd991b3969
- Create a new course.
- In the course page, select a Certificate Template in the sidebar.
- Save and make sure it saved properly.

ed3b4effe6a22c29a390b6d60669f0f9650e6f00
7415e02d3cd5f98b5b2d7efeb2eb3c10b140bb1c
- Go to learner profile page ( http://{SENSEI_URL}/learner/{USER_NAME}/ ).
- Toggle the “Allow my Certificates to be publicly viewed” checkbox.
- Save and make sure it saved properly.
- Also make sure the feedback message appears correctly.


### Deprecated Code

* `WooThemes_Sensei_Certificates::add_inline_js`
* `WooThemes_Sensei_Certificates::output_inline_js`
* `certificate_templates_wp_color_picker_js`

### Template
- `templates/single-certificate_template.php`
